### PR TITLE
[MRG+1] Clean up and move accented text example

### PR DESCRIPTION
**Pull request by dstansby**
I thought I'd kick off the docathon by moving and cleaning the first example in `pylab_examples`.

### CODE REVIEW

File: examples/text_labels_and_annotations/accented_text.py [hunk 1]
```diff
@@ -1,21 +1,24 @@
 r"""
-matplotlib supports accented characters via TeX mathtext
+=================================
+Using accented text in matplotlib
+=================================
 
+Matplotlib supports accented characters via TeX mathtext.
```
Comment (QuLogic) [inline examples/text_labels_and_annotations/accented_text.py]: Also supported through regular Unicode characters, though that's not shown here.

Comment (tacaswell) [inline examples/text_labels_and_annotations/accented_text.py]: I am :+1: on making this file utf-8 encoded and adding some accented characters.

Comment (anntzer) [inline examples/text_labels_and_annotations/accented_text.py]: Perhaps merge it with `unicode_demo.py` then?  See also #8083.

**Comment by dstansby** [comment]
👍 done.

nb. I am just ignoring `backend_driver` for now, since one of my other PRs will get rid of needing to specify tests one by one and manually keeping track of them.

File: examples/text_labels_and_annotations/accented_text.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+from __future__ import unicode_literals
+import matplotlib.pyplot as plt
+
+# Mathtext demo
+fig, ax = plt.subplots()
+ax.plot(range(10))
+ax.set_title(r'$\ddot{o}\acute{e}\grave{e}\hat{O}'
+             r'\breve{i}\bar{A}\tilde{n}\vec{q}$', fontsize=20)
+
+# Shorthand is also supported and curly's are optional
```
Comment (anntzer) [inline examples/text_labels_and_annotations/accented_text.py]: "curlies" perhaps?

File: examples/text_labels_and_annotations/accented_text.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ax.set_xlabel(r"""$\"o\ddot o \'e\`e\~n\.x\^y$""", fontsize=20)
+fig.tight_layout()
+
+# Unicode demo
+fig, ax = plt.subplots()
+ax.set_title('Développés et fabriqués')
+ax.set_xlabel("réactivité nous permettent d'être sélectionnés et adoptés")
```
Comment (anntzer) [inline examples/text_labels_and_annotations/accented_text.py]: Actually both French sentences are meaningless.  If we want to keep some foreign languages, I can propose
```
GISCARD CHAHUTÉ À L'ASSEMBLÉE
LE COUP DE DÉ DE DE GAULLE
```
(famous examples where keeping accents on caps is needed for the sentence to make any sense) :-)

Comment (tacaswell) [inline examples/text_labels_and_annotations/accented_text.py]: Buffalo buffalo buffalo buffalo ?

Comment (anntzer) [inline examples/text_labels_and_annotations/accented_text.py]: I guess English would benefit from diacritics too...

Comment (dstansby) [inline examples/text_labels_and_annotations/accented_text.py]: Thanks for the suggestions!

File: examples/text_labels_and_annotations/accented_text.py (hunk 2 above)
Comment (anntzer) [inline examples/text_labels_and_annotations/accented_text.py]: attn @dstansby

Comment (phobson) [inline examples/text_labels_and_annotations/accented_text.py]: curly braces?

**Pull request merged**